appserver/ejb/ejb-container/src/main/java/org/glassfish/ejb/startup/EjbDeployer.java

```diff
@@ -571,33 +573,37 @@ private void createAutomaticPersistentTimersForEJB(EjbDescriptor ejbDescriptor,
     }
 
     private String getOwnerId(String target) {
-        // If target is a cluster, replace it with the instance
-        ReferenceContainer ref = domain.getReferenceContainerNamed(target);
-
-        if(ref != null && ref.isCluster()) {
-            Cluster cluster = (Cluster) ref; // guaranteed safe cast!!
-            List<Server>  instances = cluster.getInstances();
-
-            // Try a random instance in a cluster
-            int useInstance = random.nextInt(instances.size());
-            Server s0 = instances.get(useInstance);
-            if (s0.isRunning()) {
-                return s0.getName();
+        // If target is a cluster or deployment group replace it with the instance
+        List<Server> instances = Collections.EMPTY_LIST;
+        Cluster c = domain.getClusterNamed(target);
```

Comment:
Best to avoid single letter variable names or in general short uncommon abbreviations. Best to use "cluster" here instead of "c".